frontend/document/documentregistry.lua

```diff
@@ -179,15 +182,19 @@ function DocumentRegistry:showSetProviderButtons(file, filemanager_instance,  ui
                             UIManager:close(self.set_provider_dialog)
                         end,
                     })
-                end,
-            },
-        })
-        -- little trick for visual separation
-        table.insert(buttons, {})
-    end
+                end
+
+                -- just once
```

Comment:
Should probably be some kind of if elseif else construction instead. Potential lurking memory leak ?

Reply:
Yes, it feels strange without any if/elseif/else, or no return in the if after the UIManager:show() :)